internal/client/uploadservice/uploadservice.go

```diff
@@ -0,0 +1,205 @@
+package uploadservice
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"mime/multipart"
+	"net/http"
+	"net/url"
+	"os"
+
+	healthcheck "github.com/ONSdigital/dp-api-clients-go/v2/health"
+	health "github.com/ONSdigital/dp-healthcheck/healthcheck"
+	dphttp "github.com/ONSdigital/dp-net/http"
+	dprequest "github.com/ONSdigital/dp-net/request"
+	"github.com/ONSdigital/log.go/v2/log"
+)
+
+const service = "upload-service"
+
+// Client is an import api client which can be used to make requests to the API
+type Client struct {
+	cli dphttp.Clienter
+	url string
+}
+
+// New creates new instance of Client with a give import api url
+func New(url string) *Client {
+	hcClient := healthcheck.NewClient(service, url)
+
+	return &Client{
+		cli: hcClient.Client,
+		url: url,
+	}
+}
+
+// ErrInvalidAPIResponse is returned when the api does not respond with a valid status
+type ErrInvalidAPIResponse struct {
+	actualCode int
+	uri        string
+	body       string
+}
+
+// Error should be called by the user to print out the stringified version of the error
+func (e ErrInvalidAPIResponse) Error() string {
+	return fmt.Sprintf(
+		"invalid response: %d from %s: %s, body: %s",
+		e.actualCode,
+		service,
+		e.uri,
+		e.body,
+	)
+}
+
+// Code returns the status code received from the api if an error is returned
+func (e ErrInvalidAPIResponse) Code() int {
+	return e.actualCode
+}
+
+var _ error = ErrInvalidAPIResponse{}
+
+type UploadJob struct {
+	ResumableFilename    string
+	IsPublishable        bool
+	CollectionId         string
+	Title                string
+	ResumableTotalSize   int32
+	ResumableType        string
+	Licence              string
+	LicenceUrl           string
+	ResumableChunkNumber int32
+	ResumableTotalChunks int32
+	File                 *os.File
+}
+
+// Checker calls api health endpoint and returns a check object to the caller.
+func (c *Client) Checker(ctx context.Context, check *health.CheckState) error {
+	hcClient := healthcheck.Client{
+		Client: c.cli,
+		URL:    c.url,
+		Name:   service,
+	}
+
+	return hcClient.Checker(ctx, check)
+}
+
+// Upload POST's form with file to be uploaded and metadata
+func (c *Client) Upload(ctx context.Context, serviceToken string, job UploadJob) error {
+	uri := fmt.Sprintf("%s/%s", c.url, "v1/upload")
```

Comment:
just a thought :) .. any benefit making the version a parameter ?

Reply:
yeah true - i think this endpoint has changed name now anyway 🙈   ill leave - this will likely change quite a bit when we plug into the real service tbh